Early Stage (MVP)

rikashi-shifu opened the pull request:
### Please tell us what you did
1. Created basic platform
2. Created a basic search function
3. Made it responsive

### Are there any concerns that may be raised?
Haven't done rendered chords & lyrics using a necessary library


### Please add which issue does this PR close below using ("closes #issue_number")
#2

rikashi-shifu committed:
Request: Init

rikashi-shifu committed:
Request: Created the static page for searching songs with simple UI design

rikashi-shifu committed:
Request: Remake of song bank index page, added the functionality of list & gallery view

rikashi-shifu committed:
Feat: Added navigation and added components to index.tsx & song.tsx.

rikashi-shifu committed:
Feat: Value from index.tsx input will be pass to search.tsx input on submit, and create dynamic routes for song/[song].tsx

rikashi-shifu committed:
Feat: Fetch real data from backend and redo the UI for ListView & GalleryView

rikashi-shifu committed:
Feat: Partially done the UI for [song].tsx

rikashi-shifu committed:
Feat: Made it responsive & created basic search function

ETCasual commented on packages/frontend/src/pages/song/[song].tsx:
```diff
@@ -0,0 +1,145 @@
+  const [songList, setSongList] = useState<Song[]>([]);
+
+  useEffect(() => {
+    const loadSongList = async () => {
```
use `anonymous function` and `.then()` to avoid race conditions

ETCasual commented on packages/frontend/src/pages/song/[song].tsx:
```diff
@@ -0,0 +1,145 @@
+    void loadSongList();
+  }, []);
+
+  const getVideoId = (url: string) => {
```
use `[song]` as `router.query.song` instead

rikashi-shifu committed:
Feat:
Fetch data using void function, used markdown-it-chords library to render the lyrics & chords properly

fgacyc reviewed (approved):
Approve for now
